src/jobs/checkExpiredPasswords.ts

```diff
@@ -0,0 +1,35 @@
+import cron from "node-cron";
+import { Status } from "../interfaces";
+import { findAllUsersService } from "../services";
+import { sendExpiredPasswordNotice } from "../services/mail/sendExpiredPasswordNotice";
+
+export const checkExpiredPasswords = async (): Promise<void> => {
+  cron.schedule("30 17 */90 * *", async () => {
+    const currentDate = new Date();
+    console.log(
+      `🔖 Log: ${currentDate}: 📢 Checking for expired passwords every 90 days at 🕥 17:30...`,
+    );
+    try {
+      const users = await findAllUsersService();
+      users.forEach(async (user) => {
+        if (user.lastPasswordUpdate) {
+          if (user.lastPasswordUpdate < new Date()) {
+            user.status = Status.Needs_Password_Reset;
+            user.save();
+            const { surname, given_name, email } = user;
+            await sendExpiredPasswordNotice({ surname, given_name, email });
+          }
+        }
+      });
+      console.log(`🍏 Done checking for expired passwords.`);
```

Comment:
@shyakadavis Great Work here, however this cron job might cause some performance issues once there is a large number of users in the database. I would suggest you approach this in the following way:

1. Instead of retrieving all `users` from the database every 90 days, consider adding a `last_password_reset` timestamp column to the `users table` and update it whenever a user changes their password.
2.  Modify `findAllUsersService` function to retrieve only users whose `last_password_reset` timestamp is at least 90 days old. This will reduce the number of users that need to be processed and improve the performance of the cron job.
3. Use `Promise.all()` handle updates and notify users concurrently. 

Code suggested below:

```suggestion
      const expiredUsers = await findUsersWithExpiredPasswords();
      if (expiredUsers.length) {
        await Promise.all([
          updateUsersStatusToNeedsPasswordReset(expiredUsers),
          sendExpiredPasswordNotices(expiredUsers),
        ]);
      }
      console.log(`🍏 Done checking for expired passwords.`);
```

You will notice that you will need to create a few functions to ease the process, here is a list of those functions and what they will do. 

1. `findUsersWithExpiredPasswords()`  - function to find users with expired passwords.
```
const findUsersWithExpiredPasswords = async () => {
  const ninetyDaysAgo = new Date();
  ninetyDaysAgo.setDate(ninetyDaysAgo.getDate() - 90);
  // Remaining Logic Here
};
```

2. `updateUsersStatusToNeedsPasswordReset()`  - function to update users with expired passwords.
```
const updateUsersStatusToNeedsPasswordReset = async (users) => {
  const userIds = users.map((user) => user.id);
  return await updateUserStatusService(userIds, Status.Needs_Password_Reset);
};
```

3. `sendExpiredPasswordNotices()`  - function to handle notifications.
```
const sendExpiredPasswordNotices = async (users) => {
  const notifications = users.map(({ surname, given_name, email }) =>
    sendExpiredPasswordNotice({ surname, given_name, email })
  );
  return await Promise.all(notifications);
};
```

Reply:
That column was already there.
The next suggestions were perfect because I hadn't thought that far ahead of how it would affect performance.

As an extra, I also set the column to be updated upon user creation, and set it to that current point in time.

I had previously asked @musllim to update the column while he was still working on the #